azurerm/resource_arm_dns_cname_record.go

```diff
@@ -54,112 +54,112 @@ func resourceArmDnsCNameRecord() *schema.Resource {
 				Required: true,
 			},
 
+			"etag": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
 			"tags": tagsSchema(),
 		},
 	}
 }
 
-func resourceArmDnsCNameRecordCreate(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*ArmClient)
-	rivieraClient := client.rivieraClient
+func resourceArmDnsCNameRecordCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	dnsClient := meta.(*ArmClient).dnsClient
 
-	tags := d.Get("tags").(map[string]interface{})
-	expandedTags := expandTags(tags)
-
-	createCommand := &dns.CreateCNAMERecordSet{
-		Name:              d.Get("name").(string),
-		Location:          "global",
-		ResourceGroupName: d.Get("resource_group_name").(string),
-		ZoneName:          d.Get("zone_name").(string),
-		TTL:               d.Get("ttl").(int),
-		Tags:              *expandedTags,
-		CNAMERecord: dns.CNAMERecord{
-			CNAME: d.Get("record").(string),
-		},
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	zoneName := d.Get("zone_name").(string)
+	ttl := int64(d.Get("ttl").(int))
+	eTag := d.Get("etag").(string)
+
+	record := d.Get("record").(string)
+	cnameRecord := dns.CnameRecord{
+		Cname: &record,
 	}
 
-	createRequest := rivieraClient.NewRequest()
-	createRequest.Command = createCommand
+	tags := d.Get("tags").(map[string]interface{})
+	metadata := expandTags(tags)
 
-	createResponse, err := createRequest.Execute()
-	if err != nil {
-		return fmt.Errorf("Error creating DNS CName Record: %s", err)
-	}
-	if !createResponse.IsSuccessful() {
-		return fmt.Errorf("Error creating DNS CName Record: %s", createResponse.Error)
+	props := dns.RecordSetProperties{
+		Metadata:    metadata,
```

Comment:
from what I can see, these are the same thing, but just to confirm: is `metadata` the same thing as `tags`? [The API documentation](https://docs.microsoft.com/en-us/rest/api/dns/recordsets#RecordSets_CreateOrUpdate) doesn't make it clear - but [we currently submit these as `tags`](https://github.com/jen20/riviera/blob/master/dns/create_dns_cname_recordset.go#L23)?